tests/test_data_sources.py

```diff
@@ -64,3 +64,105 @@ def test_opensky_datasource_stream(spark):
     result.show()
     assert len(result.columns) == 18  # Check schema has expected number of fields
     assert result.count() > 0  # Verify we got some data
+
+
+def test_salesforce_datasource_registration(spark):
+    """Test that Salesforce DataSource can be registered and validates required options."""
+    spark.dataSource.register(SalesforceDataSource)
+    
+    # Test that the datasource is registered with correct name
+    assert SalesforceDataSource.name() == "salesforce"
+    
+    # Test that the data source is streaming-only (no batch writer)
+    from pyspark.sql.functions import lit
+    
+    try:
+        # Try to use batch write - should fail since we only support streaming
+        df = spark.range(1).select(
+            lit("Test Company").alias("Name"),
+            lit("Technology").alias("Industry"),
+            lit(50000.0).alias("AnnualRevenue")
+        )
+        
+        df.write.format("salesforce").mode("append").save()
+        assert False, "Should have raised error - Salesforce DataSource only supports streaming"
+    except Exception as e:
+        # This is expected - Salesforce DataSource only supports streaming writes
+        error_msg = str(e).lower()
+        # The error can be about unsupported mode or missing writer
+        assert "unsupported" in error_msg or "writer" in error_msg or "not implemented" in error_msg
+
+
+def test_salesforce_datasource_stream_write(spark):
+    """Test Salesforce streaming write functionality with real credentials."""
+    import os
+    import pytest
+    import tempfile
+    import shutil
+    
+    # Check for Salesforce credentials in environment variables
+    username = os.getenv('SALESFORCE_USERNAME')
+    password = os.getenv('SALESFORCE_PASSWORD') 
+    security_token = os.getenv('SALESFORCE_SECURITY_TOKEN')
+    
```

Comment:
ditto. Let's keep the unit test lightweight. The above test case `test_salesforce_datasource_registration` should be enough.